### DISCUSSION TITLE
Proposed evolution of “business data” architecture ?

### DISCUSSION THREAD

**Discussion by abarrau** [magicbug/Cloudlog#2538] [General]
Good morning,
I wanted to offer you an architectural change.

Indeed, in a web application architecture, we tend to separate application data from business data.
Therefore, I think it would be relevant to have consistency on this data, so that it is stored in the same place.
We observe that :
- imported QSLs: /assets/qslcard/
- downloaded eQSLs: /images/eqsl_card_images/

My proposition :
- have only 1 directory "/data/" on the root directory, which would include:
    - imported QSLs: /data/qslcard/
    - downloaded eQSLs: /data/eqsl_card_images/
    - others ...: /data/other ...(?)

This would also facilitate application updates (only 1 directory to report) and makes it easier to manage backups.

I can work on these developments if you agree.
To read and for discussion,
Thanks, 
Aurélien / F4ANS

**Comment by abarrau** [comment]
Hello community,
no opinion on this proposal?
Favorable to make a change request in this direction?
Thks, Aurélien

**Reply by AndreasK79** [reply]
Hi, I guess we are not against it, but to change, we need to move files and update all references, and be sure that it works. Breaking existing installations must be avoided!

**Comment by magicbug** [comment]
Yes don't mind this idea I'd prefer perhaps calling it "Storage" the biggest thing will be code to move existing files around into the new structure.

**Comment by abarrau** [comment]
Peter, 
I have already started to develop the functionality for esql and can be generalized to other "data" (qsl, diploma (project under study)).
A variable in the config file allows you to define the main location (ok for "Storage").
Then each functionality (esql, qsl, ..) has its own name.
I added a migration function (moving files via the interface).
I haven't pushed the config yet, because I was finishing my dev on the "contests" ;)

**Comment by abarrau** [comment]
Hi, 
PR #2652 was uploaded ;)
73

**Comment by abarrau** [comment]
hello, 
I understand this PR is not accepted. 
I close it ?
